lib/core/util.js

```diff
@@ -10,12 +10,18 @@ function isStream (body) {
   return !!(body && typeof body.on === 'function')
 }
 
+function isIterable (obj) {
+  return !!(obj != null && (typeof obj[Symbol.iterator] === 'function' || typeof obj[Symbol.asyncIterator] === 'function'))
```

Comment:
```suggestion
  return !!(obj != null && typeof obj[Symbol.iterator] === 'function')
```
Without async support in `write` method I don't think `asyncIterators` will work any longer

Reply:
async iterators should work